src/main/kotlin/org/rust/ide/utils/StructFieldsExpander.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.utils
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ref.deepResolve
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+
+fun expandStructFields(factory: RsPsiFactory, patStruct: RsPatStruct) {
+    val declaration = patStruct.path.reference.deepResolve() as? RsFieldsOwner ?: return
+    val hasTrailingComma = patStruct.rbrace.getPrevNonCommentSibling()?.elementType == RsElementTypes.COMMA
+    patStruct.dotdot?.delete()
+    val existingFields = patStruct.patFieldList
```

Comment:
Does this `patFieldList` return `_` patterns as well?

Reply:
> Does this `patFieldList` return `_` patterns as well?

Yes, this `patFieldList` also includes underscores, but the `patTupleStruct.childrenOfType<RsPatIdent>()` that was used below does not do this. That's why `_` were considered wrong fields. 

Thank you for the report!

Reply:
> but the patTupleStruct.childrenOfType<RsPatIdent>() that was used below does not do this. 

Oh! I didn't catch that. Thanks for your fix!